### ISSUE TITLE
[Feature Request]: Latent Profile Analysis (LPA)

### ISSUE THREAD

**Issue by peteroma** [jasp-stats/jasp-issues#1505]
### Description

Latent Profile Analysis (LPA)

### Purpose

LPA is a categorical latent variable modeling approach that focuses on identifying latent subpopulations within a population based on a specified set of variables.  Unlike Latent Class Analysis (LCA), LPA can work with continuous variables and variables of differing scales.

### Use-case

Predict performance based on array/configuration of biological, behavioral, and cognitive measures.  For example, pre-mission pattern of response to stress (cortisol, heart rate variability) and cognitive test (memory, reaction time) predict in-mission behavior, performance, and resilience.

### Describe the solution you would like

A module or option within a module (e.g., Classifcation) for conducting Latent Profile Analysis (LPA) based on R packages [mclust] and [TidyLPA]

### Describe alternatives that you have considered

MPLUS software is expensive, clunky, and requires coding.  If users have to code, then we might as well use R/RStudio with the mclust and TidyLPA packages, but even they're not very intuitive.  I reached out to the developers of those packages to see if they were interested in making a Jamovi module for LPA, but no luck.

### Additional context

If LPA were something JASP could incorporate, a very valuable feature would be the ability to add the profile/class number to the dataset, thus allowing comparison of other variables by profile/class.  The Jamovi modules snowRMM with Latent Class Analysis (LCA) and the k-means clustering analysis both have this feature.

Thank you for your consideration, keep up the good work!

**Comment by TarandeepKang** [comment]
I would wholeheartedly second this request!

R users often perform latent class analysis using this package:
Linzer, D. A., & Lewis, J. B. (2011). poLCA: An R Package for Polytomous Variable Latent Class Analysis. Journal of Statistical Software, 42(10), 1–29. https://doi.org/10.18637/jss.v042.i10

And latent profile analysis with:

Rosenberg, J. M., Beymer, P. N., Anderson, D. J., Lissa, C. j van, & Schmidt, J. A. (2019). tidyLPA: An R Package to Easily Carry Out Latent Profile Analysis (LPA) Using Open-Source or Commercial Software. Journal of Open Source Software, 3(30), 978. https://doi.org/10.21105/joss.00978


The critical difference between these methods is that the former is used with categorical indicator variables, and the latter with continuous indicators.

**Comment by Kucharssim** [comment]
Thank you for the request. I am keeping an eye on mixture models in general, but I am not working on it at the moment as other things have higher priority. I will discuss with @EJWagenmakers internally when could we fit that project in the timeline.

**Comment by peteroma** [comment]
Thank you for the consideration, no doubt LPA would be a welcome addition to the JASP toolbox!  If you decide to take it on, I'd be very happy to help with beta testing.

**Comment by MTom68** [comment]
Just a short note in favor of LPA / LCA: these are increasingly popular methods in psychology where I work. It would be really great to have them in JASP since other similar types of analysis (e.g., various forms of cluster analysis) are treated as second class in leading journals. Thanks for considering this request.

**Comment by peteroma** [comment]
@MTom68, thank you for the support, let's hope JASP takes this on!  Unfortunately, neither JASP nor Jamovi have capability for LPA w continuous variables.  But if you're working with ordinal survey data, the Jamovi module snowRMM has a very nice Latent Class Analysis function.

**Comment by MTom68** [comment]
Hello, thanks for giving this attention.
Best, Tamás

Prof. Dr. Tamás Martos, Ph.D.
clinical health psychologist
Institute of Psychology
University of Szeged
https://www.researchgate.net/profile/Tamas_Martos
+36 30 9649935


peteroma ***@***.***> ezt írta (időpont: 2022. febr. 9., Sze,
17:36):

**Comment by peteroma** [comment]
Hello @Kucharssim and @EJWagenmakers, just checking to see if JASP has given any more consideration to a Latent Profile Analysis (LPA) function.

**Comment by peteroma** [comment]
Hello @Kucharssim and @EJWagenmakers, just checking to see if JASP has given any more consideration to a Latent Profile Analysis (LPA) function.

**Comment by Kucharssim** [comment]
Dear @peteroma, sorry for the tardy response!

Unfortunately there is currently a lot of features we need to implement that have a higher priority. I added this issue to a plan for the 0.18 release, as that is realistically (even though still slightly optimistically) the earliest we can implement some basic functionality. 

I appreciate your enthusiasm for this feature (I share it with you!). I will make an update when we have a draft of this module (as you indicated you would be willing to give us your feedback - which we appreciate very much!)

**Comment by peteroma** [comment]
Thank you @Kucharssim and @EJWagenmakers!  We appreciate the interest and enthusiasm, and whether 0.18, 0.20, or 0.30 I'm sure will be worth the wait.  In the meantime, I am indeed very happy to help test an LPA module at any level of development (and suspect @TarandeepKang and @MTom68 may also be willing as well).  Onward & Upward!

**Comment by TarandeepKang** [comment]
Indeed, I am happy test forthcoming LCA/LPA functionality! If it helps the team, I'm in.

**Comment by MTom68** [comment]
Also from my part!
Best, Tamás

TarandeepKang ***@***.***> ezt írta (időpont: 2022. ápr. 6.,
Sze, 11:14):

**Comment by EJWagenmakers** [comment]
@Kucharssim: Yes, this would be really nice to have!

**Comment by peteroma** [comment]
Colleagues @MTom68 @TarandeepKang, I'm excited to report that the Jamovi module snowRMM now includes Latent Profile Analysis function based on R tidyLPA and mclust (https://forum.jamovi.org/viewtopic.php?f=2&t=2168).  Would still love to see it in JASP, but it's nice to have.

**Comment by MTom68** [comment]
Great news, thanks - I checked it, and it's basic but functional.
And yes, it would be great to have it in Jasp too :).
Best, Tamás


peteroma ***@***.***> ezt írta (időpont: 2022. aug. 10., Sze,
23:03):

**Comment by peteroma** [comment]
Hello @Kucharssim and @EJWagenmakers, just checking to see if JASP has given any more consideration to a Latent Profile Analysis (LPA) function.

**Comment by peteroma** [comment]
Hello @Kucharssim and @EJWagenmakers, just checking to see if JASP has given any more consideration to a Latent Profile Analysis (LPA) function.

**Comment by TarandeepKang** [comment]
Hi team, just wondering if there has been any development with regard to this feature? It, together with  #2032 would form a very useful module!
